Tests/UnitTests/MockEdgeResponseHandler.swift

```diff
@@ -16,9 +16,28 @@ import Foundation
 class MockEdgeResponseHandler: EdgeResponseHandler {
     var onResponseReceivedData: [String: Any] = [:] // latest data received in the onResponse callback
     var onResponseCalledTimes = 0 // the number of times onResponse was called
+    var onErrorUpdateData: EdgeEventError?// latest error received in the onErrorUpdate callback
+    var onErrorUpdateCalledTimes = 0 // the number of times onResponse was called
+    var onCompleteCalledTimes = 0 // the number of times onComplete was called
 
-    func onResponse(data: [String: Any]) {
+    func onResponseUpdate(eventHandle: EdgeEventHandle) {
         onResponseCalledTimes += 1
-        onResponseReceivedData = data
+        onResponseReceivedData = eventHandle.asDictionary() ?? [:]
+    }
+
+    func onErrorUpdate(error: EdgeEventError) {
+        onErrorUpdateCalledTimes += 1
+        onErrorUpdateData = error
+    }
+
+    func onComplete() {
+        onCompleteCalledTimes += 1
+    }
+}
+
+extension EdgeEventHandle {
+    func toDictionary() -> [String: Any]? {
```

Comment:
I've seen this implemented a few times. Any way we can have a single source of truth for this function?